lib/mayaUsd/ufe/UsdUndoDeleteCommand.cpp

```diff
@@ -47,6 +47,28 @@ bool hasLayersMuted(const PXR_NS::UsdPrim& prim)
     }
     return false;
 }
+
+void removePrimRecursive(const PXR_NS::UsdPrim& prim, const PXR_NS::UsdStageWeakPtr& stage)
```

Comment:
I might be misunderstanding the problem this is trying to solve.  Why is RemovePrim() of the root of the sub-graph insufficient?

I thought the problem we were trying to solve was a list of prims passed to the delete command, either through the command line or through selection, where a parent appears in the list before a child.  In such a case, won't RemovePrim() of the child still fail, as it will be a stale prim from the moment where the parent has been RemovePrim()'ed?

Reply:
There were a few problems I was chasing to make this work...but I'm glad you asked since recursive prim removal wasn't finally necessary. It's removed now.